src/syft/core/io/route.py

```diff
@@ -119,9 +119,22 @@ def send_msg_without_reply(self, msg:SyftMessageWithoutReply) -> None:
 
 class BroadcastRoute(Route):
 
-    def send_msg_with_reply(self, msg:SyftMessageWithReply) -> Set[SyftMessageWithoutReply]:
-        raise NotImplementedError
+    def __init__(self, topic: str, connection: ClientConnection):
+        self.connection = connection
+        self.topic = topic
+
+    def send_msg_with_reply(self, msg):
+        self.connection.send_msg_with_reply(topic = self.topic, msg = msg)
+
+    def send_msg_without_reply(self, msg):
+        self.connection.send_msg_without_reply(topic = self.topic, msg = msg)
```

Comment:
topic should not be an argument of connection.send_msg_wtihout_reply. It should be a part of the address within the message.